### PR TITLE
Upgrade to go-ethereum v1.8.15

### PR DESCRIPTION
**Pull request by j16r**
[Finishes #159902922]

### CODE REVIEW

**Comment by j16r** [comment]
Not working on CI for some reason, investigating...

File: store/eth_client.go [hunk 1]
```diff
@@ -157,7 +157,9 @@ type TxReceipt struct {
 	Hash        common.Hash `json:"transactionHash"`
 }
 
+var emptyHash = common.Hash{}
```
Comment (NavyAdmiral) [inline store/eth_client.go]: 🚀

File: store/eth_client.go [hunk 2]
```diff
@@ -157,7 +157,9 @@ type TxReceipt struct {
...
+
 // Unconfirmed returns true if the transaction is not confirmed.
 func (txr *TxReceipt) Unconfirmed() bool {
-	return common.EmptyHash(txr.Hash)
+	return txr.Hash == emptyHash
```
Comment (rupurt) [inline store/eth_client.go]: Was `EmptyHash` removed from `go-ethereum`?

Comment (j16r) [reply store/eth_client.go]: Yup.

**Pull request merged**